fplus-database/src/database/applications.rs

```diff
@@ -323,7 +355,29 @@ pub async fn update_application(
             active_application.sha = Set(Some(file_sha));
             if let Some(path) = path {
                 active_application.path = Set(Some(path));
-            }
+            };
+            let updated_application = active_application.update(&conn).await?;
+            Ok(updated_application)
+        }
+        Err(_) => Err(sea_orm::DbErr::Custom(
+            "Failed to find the application to update.".into(),
+        )),
+    }
+}
+
+pub async fn update_application_issue_number(
+    owner: String,
+    repo: String,
+    pr_number: u64,
+    issue_number: Option<i64>,
+) -> Result<ApplicationModel, sea_orm::DbErr> {
+    let conn = get_database_connection().await?;
+
+    match get_application_by_pr_number(owner.clone(), repo.clone(), pr_number).await {
```

Comment:
What about the case when there's both merged and open pr application rows in DB? Will  both get updated or only the PR one?